OSDOCS-2329: GCP Workload Identity (#42544)
openshift/openshift-docs: modules/cco-ccoctl-creating-at-once.adoc

This command can take a few moments to run.
====

ifdef::aws-sts,google-cloud-platform[]
. Use the `ccoctl` tool to process all `CredentialsRequest` objects in the `credrequests` directory:
+
endif::aws-sts,google-cloud-platform[]
ifdef::aws-sts[]
[source,terminal]
----
$ ccoctl aws create-all --name=<name> --region=<aws_region> --credentials-requests-dir=<path_to_directory_with_list_of_credentials_requests>/credrequests
----
+
where:
+
** `<name>` is the name used to tag any cloud resources that are created for tracking.
** `<aws_region>` is the AWS region in which cloud resources will be created.
** `<path_to_directory_with_list_of_credentials_requests>/credrequests` is the directory containing the files for the component `CredentialsRequest` objects.
endif::aws-sts[]
ifdef::google-cloud-platform[]
[source,terminal]
----
$ ccoctl gcp create-all --name=<name> --region=<gcp_region> --project=<gcp_project_id> --credentials-requests-dir=<path_to_directory_with_list_of_credentials_requests>/credrequests
----
+
where:
+

> ** `<name>` is the name used to tag any cloud resources that are created for tracking.

Reviewer: We do not tag these resources in GCP. It is all name prefix.
```suggestion
** `<name>` is the user-defined name for all created Google cloud resources used for tracking.
```

** `<gcp_region>` is the GCP region in which cloud resources will be created.
** `<gcp_project_id>` is the GCP project ID in which cloud resources will be created.

> ** `<path_to_directory_with_list_of_credentials_requests>/credrequests` is the directory containing the files for the component `CredentialsRequest` objects.

Reviewer: ```suggestion
** `<path_to_directory_with_list_of_credentials_requests>/credrequests` is the directory containing the files of CredentialsRequests manifests to create gcp service accounts.
```
